File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NewGuidShouldNotBeUsed.vb
Imports System

Public Class NewGuid

    Public Sub WithoutArguments()

        Dim id As Guid = New Guid()  ' Noncompliant {{Use 'Guid.NewGuid()' or 'Guid.Empty' or add arguments to this Guid instantiation.}}
        '                ^^^^^^^^^^
    End Sub

    Public Sub WithArguments()
        Dim id As Guid = New Guid(New Byte() {}) ' Compliant
    End Sub

    Public Sub Other()
        Dim empty As Guid = Guid.Empty ' Compliant
        Dim rnd As Guid = Guid.NewGuid() ' Compliant
    End Sub

End Class

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NewGuidShouldNotBeUsed.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Tests.Diagnostics
{
    public class Program
    {
        public void Foo()
        {
            var g1 = new Guid(); // Noncompliant {{Use 'Guid.NewGuid()', 'Guid.Empty' or the constructor with arguments.}}
//                   ^^^^^^^^^^

            g1 = Guid.Empty;
            g1 = Guid.NewGuid();
            g1 = new Guid(new byte[0]);

            var g2 = default(Guid); // Noncompliant
//                   ^^^^^^^^^^^^^
            var guidQualifiedName = default(System.Guid); // Noncompliant

            var anotherTest = new StringBuilder(); // Checking that the rule raises issue only for the Guid new Object instatiation.
            var anotherTestDefault = default(long);
            var anotherTestDefaultQualifiedName = default(System.String);
        }

        public Guid Get() => default(Guid); // Noncompliant

        private T Test<T>() where T : new()
        {
            return new T(); // Covering the case where the MethodSymbol has not Type
        }
    }
}
